pkg/pgmodel/migrations/sql/preinstall/003-tables.sql

```diff
@@ -23,9 +23,14 @@ INSERT INTO public.prom_installation_info(key, value) VALUES
 CREATE TABLE SCHEMA_CATALOG.series (
     id bigint NOT NULL,
     metric_id int NOT NULL,
-    labels SCHEMA_PROM.label_array NOT NULL --labels are globally unique because of how partitions are defined
+    labels SCHEMA_PROM.label_array NOT NULL, --labels are globally unique because of how partitions are defined
+    delete_epoch bigint -- epoch after which this row can be deleted
 ) PARTITION BY LIST(metric_id);
 CREATE INDEX series_labels_id ON SCHEMA_CATALOG.series USING GIN (labels);
+CREATE INDEX series_deleted
+    ON SCHEMA_CATALOG.series
+    USING GIN (labels) -- no default operator for bigint so no epoch
```

Comment:
I was thinking this index would be needed for either the deadness scan, or the label read, but looking at it now I don't think it is

Reply:
our WHERE clauses
```
WHERE delete_epoch IS NOT NULL
            AND labels = (SELECT * FROM cte)

WHERE delete_epoch IS NULL
                AND series_id IN (SELECT * FROM confirmed_drop_series)

WHERE delete_epoch <= deletion_epoch
    AND NOT EXISTS (
        SELECT 1 FROM SCHEMA_DATA.%1$I
        WHERE id = label_id LIMIT 1
    )

WHERE delete_epoch <= deletion_epoch
                    AND id IN (SELECT id FROM dead_series) -- (I'm not sure both this and the above is needed
```

Reply:
The first and second ones must be driven based on `series_labels_id` because they're looking for `NULL`s. The third and fourth should be driven by `delete_epoch`, and having `id` could change them to and IndexOnly scans.

I think this means the index should be btree `(delete_epoch, id)`

Reply:
in my experiment, it looks like the index gets created
```SQL
mat=> create table test_table_root(i int, j int) partition by list (i);
CREATE TABLE

mat=> create index on test_table_root (i) where j is not null;
CREATE INDEX

mat=> create table test_table_child(i int, j int);
CREATE TABLE

mat=> ALTER TABLE test_table_root ATTACH PARTITION test_table_child FOR VALUES IN (1);
ALTER TABLE

mat=> \d+ test_table_root
                        Partitioned table "public.test_table_root"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 i      | integer |           |          |         | plain   |              | 
 j      | integer |           |          |         | plain   |              | 
Partition key: LIST (i)
Indexes:
    "test_table_root_i_idx" btree (i) WHERE j IS NOT NULL
Partitions: test_table_child FOR VALUES IN (1)

mat=> \d+ test_table_child
                             Table "public.test_table_child"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 i      | integer |           |          |         | plain   |              | 
 j      | integer |           |          |         | plain   |              | 
Partition of: test_table_root FOR VALUES IN (1)
Partition constraint: ((i IS NOT NULL) AND (i = 1))
Indexes:
    "test_table_child_i_idx" btree (i) WHERE j IS NOT NULL
Access method: heap
```